### PR TITLE
Improve DuelOverview and Avatar usage

### PR DESCRIPTION
**Pull request by nathanaelhoun**
## Changes

- [x] Business logic should not go in the component's `render()` method, to prevent recalculation of all the informations at each rendering. Switched business logic of duelOverview to the data-fetching and cleaned up the code.
- [x] We don't often change only one part of an Avatar, so it's way easier to group all the properties into one object and swap the whole object when we want to update only one property of the avatar (you were right @FrancoisPog). Makes `<Avatar/>` way easier to use.
- [x] Original aim of the PR: clarify the "play round" button on the duelOverview and remove it if the duel is finished

image
 
image

### CODE REVIEW

File: client/src/styles/components/_quiz.scss [hunk 1]
```diff
@@ -179,3 +179,20 @@ $timer-size: 100px;
     height: 100%;
   }
 }
+
+/* ---------- DuelResults ---------- */
+
+.duel-results {
```
Comment (valentinperignon) [inline client/src/styles/components/_quiz.scss:185]: Why keeping the button format?
It's a good idea to change the text to "win / lose" but having this big red or green banner is not great... It makes a big color block

File: client/src/pages/Profile.js [hunk 2]
```diff
@@ -79,6 +82,10 @@ export default class Profile extends Component {
         <button className="btn" onClick={this.handleLogoutClick}>
           Me déconnecter
         </button>
+
+        <Link to="/about" className="btn btn-fw">
```
Comment (valentinperignon) [inline client/src/pages/Profile.js:86]: Why a full-width button button ? There are three buttons in this page, three different sizes

File: client/src/components/Avatar.js [hunk 3]
```diff
@@ -1,55 +1,51 @@
...
+import { ReactComponent as BodyBottomImage } from "../images/avatar/bodyBottom.svg";
 import ImageBodyTop from "../images/avatar/bodyTop.png";
 import ImageEyes from "../images/avatar/eyes.png";
 import ImageHands from "../images/avatar/hands.png";
 import ImageHats from "../images/avatar/hats.png";
 import ImageMouthes from "../images/avatar/mouthes.png";
 
-const Avatar = function ({ size, eyes, hands, hat, mouth, colorBG, colorBody, reversed }) {
-  return (
+const Avatar = ({ size, infos: { eyes, hands, hat, mouth, colorBG, colorBody }, reversed }) => (
```
Comment (valentinperignon) [inline client/src/components/Avatar.js:11]: Great 👍

**Review by valentinperignon — COMMENTED** [review]
Great work! grrrr

File: client/src/pages/Profile.js (hunk 2 above)
Comment (nathanaelhoun) [reply client/src/pages/Profile.js:86]: It's mainly because I copy-pasted it. It's fixed!

File: client/src/styles/components/_quiz.scss (hunk 1 above)
Comment (nathanaelhoun) [reply client/src/styles/components/_quiz.scss:185]: I quickly prototyped it as I was thinking about using a new _Pilette_ design to illustrate this part of the app. For the moment, I have applied your suggestions (and i looks much better :smile:)

<details>
  <summary></summary>
  The button is gone but something has replaced it! Grrrrrr
</details>

image
image

**Review by valentinperignon — APPROVED** [review]
LGTM!

**Review by FrancoisPog — APPROVED** [review]
LGTM !

**Pull request merged**